### ISSUE TITLE
Report full path in path errors?

### ISSUE THREAD

**Issue by professionalPeter** [JohnSundell/Unbox#165]
Have there been attempts to provide the full path in error messages for object hierarchies? Is there a known reason that this can't or shouldn't be done? 

For example, given the following JSON:
```
{
  "root": {
    "foo": [
      {
        "bar": 2
      },
      {
        "ImABadKey": 2
      }
    ],
  "fish": "yum"
  }
}
```

and the following model code:
```
struct Root: Unboxable {
    let model: Model

    init(unboxer: Unboxer) throws {
        model = try unboxer.unbox(key: "root")
    }

}

struct Model: Unboxable {
    let foos: [Foo]
    let fish: String

    init(unboxer: Unboxer) throws {
        foos = try unboxer.unbox(key: "foo")
        fish = try unboxer.unbox(key: "fish")
    }
}

struct Foo: Unboxable {
    let bar: String

    init(unboxer: Unboxer) throws {
        bar = try unboxer.unbox(key: "bar")
    }
}
```

I'd love to see:
`[UnboxError] An error occured while unboxing path "bar": The key "bar" is missing at "root.foo[1]"`
instead of 
`[UnboxError] An error occured while unboxing path "bar": The key "bar" is missing.`

**Comment by professionalPeter** [comment]
BTW, thanks for a great library!

**Comment by JohnSundell** [comment]
This would be a really nice addition to the error API! 👍 Perhaps emit the `root`, since it's a given, and instead make the output path just `foo[1].bar`. Would you like to contribute this as a PR @professionalPeter?

**Comment by professionalPeter** [comment]
Yes, I'd like to, however to be honest, I'm it's unclear if/when I'll have the opportunity to. If I can, I definitely will now that I know it's not going against the direction of the project. Thanks!

**Comment by JohnSundell** [comment]
Sure, no problems! 👍
